docker-api/src/main/java/com/yahoo/vespa/hosted/dockerapi/ContainerResources.java

```diff
@@ -36,11 +41,20 @@
             throw new IllegalArgumentException("memoryBytes must be a positive integer or 0 for unlimited, was " + memoryBytes);
     }
 
+    /**
+     * Create container resources from required fields.
+     *
+     * @param cpus the amount of vcpu that should be exclusively available to this container. This is a hard limit:
+     *             More than this amlunt will never be available. To allow an unlimited amount use 0.
+     * @param cpuCores the amount of vcpu that should ideally be allocated to this container if there are no other
```

Comment:
This is a more of a guarantee of lower bound of vcpus: The container is allocated a number of shares proportional to its vcpus. These shares are only used if _host_ CPU is near fully allocated (by allocating CPU time proportional to the number of shares held by each container). Therefore, a container can go above this value, but another container cannot  starve this container.

Reply:
That makes sense, thanks. I'll rephrase.